enhance serverless vs dedicated vs byoc (#31)
redpanda-data/cloud-docs: modules/get-started/pages/cloud-overview.adoc

TIP: For more detailed information about the Redpanda platform, see xref:get-started:intro-to-events.adoc[] and xref:get-started:architecture.adoc[].

== Redpanda Cloud cluster types

> Redpanda offers three types of fully-managed cloud clusters: Serverless, Dedicated, and BYOC.

Reviewer: As we expand or revise these offerings, may make more sense not to specify the number and instead say:
Redpanda offers the following fully-managed cloud cluster types:

|===
| Serverless | Dedicated | Bring Your Own Cloud (BYOC)

| Hosted in Redpanda Cloud | Hosted in Redpanda Cloud | Hosted in your private cloud
| Instantly available with no base cost | Provides higher performance and reliability | Provides data sovereignty + greater security and isoloation
| Cloud providers = AWS | Cloud providers = AWS, GCP | Cloud providers = AWS, GCP, Azure

> | Public networking | Private networking | Private networking

Reviewer 2: Include Public networking for Dedicated and BYOC. It feels like we only support Private.

| Multi-tenant | Single-tenant | Single-tenant
| Single-AZ | Multi-AZ | Multi-AZ
| Ingress = up to 10 MBps | Ingress = up to 400 MBps | Ingress = up to 2 GBps
| Egress = up to 30 MBps | Egress = up to 800 MBps | Egress = up to 4 GBps
| Logical partitions = up to 100 | Logical partitions = up to 22,800 | Logical partitions = up to 112,500
|  | Client connections = up to 180,000 | Client connections = up to 450,000
| Retention = 1 TB | Retention = unlimited | Retention = unlimited
| SLA = 99.50% uptime | SLA = 99.95% uptime + 24x7 monitoring and alerting | SLA = 99.95% uptime + 24x7 monitoring and alerting
| Support = Community Slack channel | Support = Dedicated Slack channel + account manager | Support = Dedicated Slack channel + account manager
|===

NOTE: With standard BYOC clusters, Redpanda manages security policies and resources for your VPC, including service accounts, subnetworks, IAM roles, firewall rules, and storage buckets. For the most security, you can manage these resources yourself with a xref:get-started:cluster-types/byoc/vpc-byo-gcp.adoc[customer-managed VPC].

=== Serverless vs Dedicated vs BYOC

> Serverless clusters are a good fit for the following:

Reviewer: ```suggestion
Serverless clusters are a good fit for the following use cases:
```

* Starter and growing workloads
* Spiky workloads (that is, development environments, systems that only occasionally get busy, or workloads that come and go)
* Fast and dynamic cluster creation: you can use a Serverless cluster as an isolated container for topics

> With Serverless (or if you sign up for Dedicated through the AWS Marketplace), you only pay for what you consume, without any commitment. A cluster is created instantly, so you can surface it in your applications (for example, for tenant isolation). If your workload increases, you can migrate it to a Dedicated or BYOC cluster.

Reviewer: ```suggestion
With Serverless (and for Dedicated when procured through the AWS Marketplace), you only pay for what you consume, without any commitment. A cluster is created instantly, so you can surface it in your applications (for example, for tenant isolation). If your workload increases, you can migrate it to a Dedicated or BYOC cluster.
```
